### PR TITLE
Update to .NET Core 2.1.4 with SDK 2.1.402

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Update to .NET Core 2.1.4 with SDK 2.1.402

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: assets/files.wxs [hunk 1]
```diff
@@ -188,7 +188,7 @@
         <File Id="fil8E8E3BF1C97D92DFEB4FC97D778F238F" KeyPath="yes" Source="$(env.ProductSourcePath)\System.Runtime.CompilerServices.VisualC.dll" />
       </Component>
       <Component Id="cmpf65fc69858294501b4f61951193b392c" Guid="{09C9310D-56AD-4F7A-90C8-DF0F7E8A7251}">
-        <File Id="filDDE80F6ACEC54B56BA33E587A072DE27" KeyPath="yes" Source="$(env.ProductSourcePath)\sos_$(var.FileArchitecture)_$(var.FileArchitecture)_4.6.26725.06.dll" />
+        <File Id="filDDE80F6ACEC54B56BA33E587A072DE27" KeyPath="yes" Source="$(env.ProductSourcePath)\sos_$(var.FileArchitecture)_$(var.FileArchitecture)_4.6.26814.03.dll" />
```
Comment (anmenaga) [inline assets/files.wxs]: @TravisEz13 does the Id on these has to be re-generated? (to be unique for servicing purposes)

Comment (TravisEz13) [reply assets/files.wxs]: either the ID or the Guid of the component need to change.  Can you not make the change manually and use the file that `Start-PSPackage` generates and tells you to use?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
See comments

File: assets/files.wxs (hunk 1 above)
Comment (daxian-dbw) [reply assets/files.wxs]: Fixed.

**Comment by daxian-dbw** [comment]
@TravisEz13 You comment is addressed. Please take another look, thanks.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**